test/functional/rpc_txospender.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the gettxospender RPC call.
+"""
+
+from test_framework.blocktools import COINBASE_MATURITY
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+    find_vout_for_address,
+)
+
+class GetTxoSpenderTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [
+            ["-txindex", "-txospenderindex"],
+            ["-txindex"],
+        ]
+        # whitelist all peers to speed up tx relay / mempool sync
+        for args in self.extra_args:
+            args.append("-whitelist=noban@127.0.0.1")
+
+        self.supports_cli = False
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
```

Comment:
RPC tests shouldn't require wallet, please use `MiniWallet`

Reply:
I've translated the test to MiniWallet. If you take it, make sure to rebase on current master first, as it is using new MiniWallet functions.

Also, I am removing not needed stuff. Let me know if you have any questions. And no need to mention me in the commit as author.

```diff
diff --git a/test/functional/rpc_txospender.py b/test/functional/rpc_txospender.py
index 1d7157e9bf..c063b4e502 100755
--- a/test/functional/rpc_txospender.py
+++ b/test/functional/rpc_txospender.py
@@ -10,57 +10,36 @@ from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import (
     assert_equal,
     assert_raises_rpc_error,
-    find_vout_for_address,
 )
+from test_framework.wallet import MiniWallet
+
 
 class GetTxoSpenderTest(BitcoinTestFramework):
     def set_test_params(self):
-        self.setup_clean_chain = True
         self.num_nodes = 2
         self.extra_args = [
             ["-txindex", "-txospenderindex"],
             ["-txindex"],
         ]
-        # whitelist all peers to speed up tx relay / mempool sync
-        for args in self.extra_args:
-            args.append("-whitelist=noban@127.0.0.1")
-
         self.supports_cli = False
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
-    def setup_network(self):
-        super().setup_network()
-        self.connect_nodes(0, 1)
-
     def run_test(self):
-        self.log.info("Prepare some coins for multiple *rawtransaction commands")
-        self.generate(self.nodes[1], 1)
-        self.generate(self.nodes[0], COINBASE_MATURITY + 1)
-        self.sync_all()
-        self.generate(self.nodes[0], 5)
+        self.wallet = MiniWallet(self.nodes[1])
+        self.wallet.rescan_utxos()
 
         self.gettxospender_tests()
 
     def gettxospender_tests(self):
-        addr = self.nodes[1].getnewaddress()
-        txid = self.nodes[0].sendtoaddress(addr, 10)
-        self.generate(self.nodes[0], 1)
-        vout = find_vout_for_address(self.nodes[1], txid, addr)
-        rawTx = self.nodes[1].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): 9.999})
-        rawTxSigned = self.nodes[1].signrawtransactionwithwallet(rawTx)
-        txId = self.nodes[1].sendrawtransaction(rawTxSigned['hex'])
-        self.generateblock(self.nodes[0], output=self.nodes[0].getnewaddress(), transactions=[rawTxSigned['hex']])
+        partially_spent_tx = self.wallet.send_self_transfer_multi(from_node=self.nodes[1], num_outputs=2)
+        num_out = 1
+        spending_txid = self.wallet.send_self_transfer(from_node=self.nodes[1], utxo_to_spend=partially_spent_tx["new_utxos"][num_out])["txid"]
+        self.generate(self.wallet, 1)
 
-        for n in [0, 1]:
-            self.log.info(f"Test gettxospender {'with' if n == 0 else 'without'} -txospenderindex")
+        self.log.info(f"Test gettxospender with -txospenderindex")
+        assert_equal(self.nodes[0].gettxospender(partially_spent_tx["txid"], num_out), spending_txid)
+        self.log.info(f"Test gettxospender without -txospenderindex")
+        assert_raises_rpc_error(-8, "This RPC call requires -txospenderindex to be enabled.", self.nodes[1].gettxospender, partially_spent_tx["txid"], num_out)
 
-            if n == 0:
-                assert_equal(self.nodes[n].gettxospender(txid, vout), txId)
-            else:
-                # Without -txospenderindex, expect to raise.
-                assert_raises_rpc_error(-8, "This RPC call requires -txospenderindex to be enabled.", self.nodes[n].gettxospender, txid, vout)
 
-if __name__ == '__main__':
+if __name__ == "__main__":
     GetTxoSpenderTest().main()

Reply:
Thanks! I've rebased on https://github.com/bitcoin/bitcoin/pull/24408 and used your code to extend its RPC test